packages/SwingSet/src/devices/vat-admin/device-vat-admin.js

```diff
@@ -194,65 +191,43 @@ export function buildDevice(tools, endowments) {
 
         // D(devices.vatAdmin).createByBundle(bundle, options) -> vatID
         if (method === 'createByBundle') {
-          // see #4588
-          assert.equal(argsCapdata.slots.length, 0, 'cannot handle refs');
-          const args = unserialize(argsCapdata);
-          const [bundle, options] = args;
-          // options cannot hold caps yet, but I expect #4486 will remove
-          // this case before #4381 needs them here
-          const vatID = pushCreateVatBundleEvent(bundle, options);
+          const res = syscall.callKernelHook('createByBundle', argsCapdata);
```

Comment:
Presumably the syscall implementation does parameter validation, but it feels weird to be passing a capdata received from the outside world directly into the kernel without looking at it.

Reply:
Yeah, `vat-vat-admin` does the user-visible-error -producing validation (in `convertOptions`), and the device basically just fronts for the kernel hook. (hence #4864 being about cutting out `device-vat-admin.js` entirely).